src/platform/qt/library/LibraryModel.h

```diff
@@ -0,0 +1,84 @@
+/* Copyright (c) 2013-2022 Jeffrey Pfau
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+#pragma once
+
+#include <QAbstractItemModel>
+#include <QIcon>
+#include <QTreeView>
+
+#include <mgba/core/library.h>
+
+#include <memory>
+#include <vector>
+
+#include "LibraryEntry.h"
+
+class QTreeView;
+
+namespace QGBA {
+
+class LibraryModel final : public QAbstractItemModel {
+Q_OBJECT
+
+public:
+	enum Columns {
+		COL_NAME = 0,
+		COL_LOCATION = 1,
+		COL_PLATFORM = 2,
+		COL_SIZE = 3,
+		COL_CRC32 = 4,
+		MAX_COLUMN = 4,
```

Comment:
Generally speaking, I prefer using an exclusive bound because you don't need to manually assign the value that way. You'll need to update the <= to < in places of course, but I find it somewhat easier to maintain. Especially if we're gonna add columns.

Reply:
I broadly agree but in terms of the implementation I have more reason to refer to the last column than to the number of columns; it's not just there for iteration. I'll make the switch if you insist but I'll be adding some `- 1`'s in places and removing `+ 1`'s in others.